src/bike.c

```diff
@@ -0,0 +1,430 @@
+            return FALSE;
+    }
+    return TRUE;
+}
+
+void StartTransitionToFlipBikeState(u8 flags)
+{
+    gBikeCameraAheadPanback = FALSE;
+    if (gPlayerAvatar.flags & (PLAYER_AVATAR_FLAG_MACH_BIKE | PLAYER_AVATAR_FLAG_ACRO_BIKE))
+    {
+        SetPlayerAvatarTransitionFlags(PLAYER_AVATAR_FLAG_ON_FOOT);
+        Overworld_ClearSavedMusic();
+        Overworld_PlaySpecialMapMusic();
+    }
+    else
+    {
+        SetPlayerAvatarTransitionFlags(flags);
+        if (sub_8056124(MUS_CYCLING))
+        {
+            Overworld_SetSavedMusic(MUS_CYCLING);
+            Overworld_ChangeMusicTo(MUS_CYCLING);
+        }
+    }
+}
+
+void sub_80BD620(u32 directionHistory, u32 abStartSelectHistory)
+{
+    u8 i;
+
+    gPlayerAvatar.acroBikeState = ACRO_STATE_NORMAL;
+    gPlayerAvatar.newDirBackup = 0;
+    gPlayerAvatar.bikeFrameCounter = 0;
+    gPlayerAvatar.bikeSpeed = SPEED_STANDING;
+    gPlayerAvatar.directionHistory = directionHistory;
+    gPlayerAvatar.abStartSelectHistory = abStartSelectHistory;
+    gPlayerAvatar.lastSpinTile = 0;
+    for (i = 0; i < NELEMS(gPlayerAvatar.dirTimerHistory); ++i)
+            gPlayerAvatar.dirTimerHistory[i] = 0;
+}
+
+void Bike_UpdateBikeCounterSpeed(u8 counter)
+{
+    gPlayerAvatar.bikeFrameCounter = counter;
+    gPlayerAvatar.bikeSpeed = counter + (gPlayerAvatar.bikeFrameCounter >> 1);
+}
+
+static void sub_80BD664(void)
+{
+    gPlayerAvatar.bikeFrameCounter = 0;
+    gPlayerAvatar.bikeSpeed = SPEED_STANDING;
+}
+
+s16 GetPlayerSpeed(void)
+{
+    s16 exp[] = { 1, 2, 4 };
+
+    if (gPlayerAvatar.flags & PLAYER_AVATAR_FLAG_MACH_BIKE)
+        return exp[gPlayerAvatar.bikeFrameCounter];
+    else if (gPlayerAvatar.flags & PLAYER_AVATAR_FLAG_ACRO_BIKE)
+        return 3;
+    else if (gPlayerAvatar.flags & (PLAYER_AVATAR_FLAG_SURFING | PLAYER_AVATAR_FLAG_WATERING))
+        return 2;
+    else
+        return 1;
+}
+
+void Bike_HandleBumpySlopeJump(void)
+{
+    s16 x, y;
+    u8 tileBehavior;
+
+    if (gPlayerAvatar.flags & PLAYER_AVATAR_FLAG_ACRO_BIKE)
+    {
+        PlayerGetDestCoords(&x, &y);
+        tileBehavior = MapGridGetMetatileBehaviorAt(x, y);
+        if (MetatileBehavior_IsBumpySlope(tileBehavior))
+        {
+            gPlayerAvatar.acroBikeState = ACRO_STATE_WHEELIE_STANDING;
+            nullsub_24(GetPlayerMovementDirection());
+        }
+    }
+}
+
+static const u8 gUnknown_83E7D22 = 4;
+
+// unused
+static const struct
+{
+    u32 unk00, unk04, unk08, unk0C;
+    const void *unk10, *unk14;
+    u32 unk18;
+} gUnknown_83E7D24[] =
+{
+    {
+        1, 2, 15, 15,
+        &gUnknown_83E7D22, &gUnknown_83E7D22,
+        1,
+    },
+    {
+        2, 2, 15, 15,
+        &gUnknown_83E7D22, &gUnknown_83E7D22,
+        2,
+    },
+    {
+        3, 2, 15, 15,
+        &gUnknown_83E7D22, &gUnknown_83E7D22,
+        3,
+    },
+    {
+        4, 2, 15, 15,
+        &gUnknown_83E7D22, &gUnknown_83E7D22,
+        4,
+    },
+};
```

Comment:
This is `sAcroBikeTricksList` in Emerald.

Reply:
Resolved.